README.md
# react-native-ignore-warnings

[![NPM](https://img.shields.io/npm/v/react-native-ignore-warnings.svg?style=flat-square)](https://www.npmjs.com/package/react-native-ignore-warnings)
[![NPM](https://img.shields.io/npm/dt/react-native-ignore-warnings.svg?style=flat-square)](https://www.npmjs.com/package/react-native-ignore-warnings)
[![GitHub stars](https://img.shields.io/github/stars/clayrisser/react-native-ignore-warnings.svg?style=social&label=Stars)](https://github.com/clayrisser/react-native-ignore-warnings)

> ⚛️🤷⚠️ Ignore react native warnings and logs

Please &#9733; this repo if you found it useful &#9733; &#9733; &#9733;

## Features

- Ignore warnings
- Ignore logs
- Ignore react native yellow box
- Compatible with Expo

## Installation

```sh
npm install --save react-native-ignore-warnings
```

## Dependencies

- [NodeJS](https://nodejs.org)
- [React Native](https://facebook.github.io/react-native)

## Usage

```ts
import ignoreWarnings from 'react-native-ignore-warnings';

ignoreWarnings(['Watch your six', 'Mind the gap', 'Beware the ides of March']);

ignoreWarnings('error', ['The house is on fire', 'System failure']);

ignoreWarnings('log', 'Hello, world');
```

## Support

Submit an [issue](https://github.com/clayrisser/react-native-ignore-warnings/issues/new)

## Contributing

Review the [guidelines for contributing](CONTRIBUTING.md)

## License

[Apache-2.0 License](LICENSE)

[Silicon Hills LLC](https://siliconhills.dev) &copy; Copyright 2018-2021

## Credits

- [Clay Risser](https://clayrisser.com) - Author


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## [0.0.1] - 2021-07-09

### Added

- Initial release


tests/index.ts
describe('react-native-ignore-warnings', () => {
  it('should work', async () => {
    expect(true).toBe(true);
  });
});

export default null;


tests/_setup.ts
/**
 * File: /tests/_setup.ts
 * Project: react-native-ignore-warnings
 * File Created: 09-07-2021 18:35:16
 * Author: Clay Risser <email@clayrisser.com>
 * -----
 * Last Modified: 09-07-2021 18:44:02
 * Modified By: Clay Risser <email@clayrisser.com>
 * -----
 * Silicon Hills LLC (c) Copyright 2021
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export default null;


.eslintrc.js
const fs = require('fs');

const cspellWords = JSON.parse(
  fs.readFileSync('.vscode/settings.json').toString()
)['cSpell.words'];

module.exports = {
  extends: ['airbnb-typescript/base', 'prettier'],
  parser: '@typescript-eslint/parser',
  root: true,
  env: {
    browser: true
  },
  plugins: ['spellcheck'],
  parserOptions: {
    project: './tsconfig.json',
    ecmaFeatures: {
      legacyDecorators: true
    }
  },
  rules: {
    '@typescript-eslint/comma-dangle': ['error', 'never'],
    '@typescript-eslint/indent': 'off',
    '@typescript-eslint/no-shadow': 'off',
    '@typescript-eslint/no-use-before-define': 'off',
    'arrow-body-style': 'off',
    'class-methods-use-this': 'off',
    'comma-dangle': 'off',
    'default-case': 'off',
    'import/extensions': ['error', 'never', { json: 'always' }],
    'import/no-cycle': 'off',
    'import/prefer-default-export': 'off',
    'max-classes-per-file': 'off',
    'max-lines': ['error', 500],
    'max-lines-per-function': ['warn', 50],
    'no-await-in-loop': 'off',
    'no-empty-function': ['warn', { allow: ['constructors'] }],
    'no-extra-boolean-cast': 'off',
    'no-param-reassign': 'off',
    'no-plusplus': 'off',
    'no-return-assign': 'off',
    'no-shadow': 'off',
    'no-underscore-dangle': 'off',
    'no-use-before-define': 'off',
    'no-useless-constructor': 'off',
    'react/jsx-props-no-spreading': 'off',
    yoda: 'off',
    'spellcheck/spell-checker': [
      'warn',
      {
        comments: true,
        strings: true,
        identifiers: true,
        lang: 'en_US',
        skipWords: cspellWords,
        skipIfMatch: ['http?://[^s]*', '^[-\\w]+/[-\\w\\.]+$'],
        skipWordIfMatch: [],
        minLength: 3
      }
    ],
    'lines-between-class-members': [
      'error',
      'always',
      { exceptAfterSingleLine: true }
    ],
    'import/no-unresolved': [
      'error',
      {
        ignore: ['^~']
      }
    ],
    'no-unused-vars': [
      'warn',
      {
        args: 'after-used',
        argsIgnorePattern: '^_',
        ignoreRestSiblings: true,
        vars: 'all'
      }
    ],
    '@typescript-eslint/no-unused-vars': [
      'warn',
      {
        args: 'after-used',
        argsIgnorePattern: '^_',
        ignoreRestSiblings: true,
        vars: 'all'
      }
    ],
    'import/no-extraneous-dependencies': [
      'error',
      {
        devDependencies: [
          '**/*.spec.js',
          '**/*.spec.jsx',
          '**/*.spec.ts',
          '**/*.spec.tsx',
          '**/*.story.js',
          '**/*.story.jsx',
          '**/*.story.ts',
          '**/*.story.tsx',
          '**/*.test.js',
          '**/*.test.jsx',
          '**/*.test.ts',
          '**/*.test.tsx',
          'storybook/**/*.js',
          'storybook/**/*.jsx',
          'storybook/**/*.ts',
          'storybook/**/*.tsx',
          'tests/*.js',
          'tests/*.jsx',
          'tests/*.ts',
          'tests/*.tsx'
        ]
      }
    ]
  },
  overrides: [
    {
      files: ['*.test.js', '*.test.jsx', '*.test.ts', '*.test.tsx'],
      env: {
        jest: true
      },
      plugins: ['jest']
    },
    {
      files: ['*.ts', '*.tsx'],
      rules: {
        'no-unused-vars': 'off'
      }
    }
  ],
  settings: {
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx']
      }
    }
  }
};


src/@types/ignoreWarnings.d.ts
/**
 * File: /src/@types/ignoreWarnings.d.ts
 * Project: react-native-ignore-warnings
 * File Created: 09-07-2021 18:37:05
 * Author: Clay Risser <email@clayrisser.com>
 * -----
 * Last Modified: 09-07-2021 18:43:19
 * Modified By: Clay Risser <email@clayrisser.com>
 * -----
 * Silicon Hills LLC (c) Copyright 2021
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

declare module 'ignore-warnings' {
  function ignoreWarnings(type: string, ignoreMessages: string | string[]): any;
  export = ignoreWarnings;
}


src/index.ts
/**
 * File: /src/index.ts
 * Project: react-native-ignore-warnings
 * File Created: 09-07-2021 18:35:16
 * Author: Clay Risser <email@clayrisser.com>
 * -----
 * Last Modified: 09-07-2021 18:43:44
 * Modified By: Clay Risser <email@clayrisser.com>
 * -----
 * Silicon Hills LLC (c) Copyright 2021
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import ignoreWarnings from 'ignore-warnings';
import { LogBox } from 'react-native';

export default function reactNativeIgnoreWarnings(
  type: string,
  ignoreMessages: string | string[]
) {
  if (!ignoreMessages) {
    ignoreMessages = type;
    type = 'warn';
  }
  if (!Array.isArray(ignoreMessages)) ignoreMessages = [ignoreMessages];
  LogBox.ignoreLogs(ignoreMessages);
  ignoreWarnings(type, ignoreMessages);
}
